src/core/lib/security/authorization/eval_args_util.cc

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_LIB_SECURITY_AUTHORIZATION_EVAL_ARGS_UTIL_CC
+#define GRPC_CORE_LIB_SECURITY_AUTHORIZATION_EVAL_ARGS_UTIL_CC
+
+#include "src/core/lib/security/authorization/eval_args_util.h"
+
+grpc::string_ref get_key(grpc_linked_mdelem* elem) {
+  const grpc_slice& key = GRPC_MDKEY(elem->md);
+  return grpc::string_ref(reinterpret_cast<const char*>(
+      GRPC_SLICE_START_PTR(key), GRPC_SLICE_LENGTH(key)));
+}
+
+grpc::string_ref get_value(grpc_linked_mdelem* elem) {
+  const grpc_slice& val = GRPC_MDVALUE(elem->md);
+  return grpc::string_ref(reinterpret_cast<const char*>(
+      GRPC_SLICE_START_PTR(val), GRPC_SLICE_LENGTH(val)));
+}
+
+grpc::string_ref get_path(grpc_metadata_batch* metadata) {
+  grpc_linked_mdelem* path = metadata->idx.array[GRPC_BATCH_PATH];
+  return get_value(path);
+}
```

Comment:
Can we use string_view(more common) here and for rest of the functions instead of string_ref? Is there any benefit of using the latter?

Something like
```
absl::string_view get_path_from_metadata(const grpc_metadata_batch* metadata) {
  absl::string_view path;
  if(metadata->idx.named.path != nullptr) {
    path = absl::string_view(reinterpret_cast<const char*>(
      GRPC_SLICE_START_PTR(GRPC_MDVALUE(metadata->idx.named.path->md))),
        GRPC_SLICE_LENGTH(GRPC_MDVALUE(metadata->idx.named.path->md)));
  }
  return path;
}
```

The snippet above has some other recommendations too
1. We want to check if the field is set in the input struct.
2. Maybe rename the functions to get_path_from_metadata? Similarly for the other functions.